lib/logstash/outputs/google_cloud_storage.rb

```diff
@@ -175,7 +175,7 @@ def receive(event)
   public
   def close
     @logger.debug('Stopping the plugin, uploading the remaining files.')
-    Stud.stop!(@registration_thread) unless @registration_thread.nil?
+    Thread.kill(@uploader_thread) unless @uploader_thread.nil?
```

Comment:
Should you also stop the registration thread?

Reply:
I replaced registration thread with uploader thread because `Stud::` was calling `kill` in the background anyway but something went wonky when doing it with the tests which left orphaned threads running.

I wish there was something that would be more fluent like:

```.ruby
Background.every 30.seconds do
	# ...
end
```